EIPS/eip-7492.md

```diff
@@ -0,0 +1,333 @@
+| spent\_     | uint256 | The amount of token being spent. This will be transfered as part of this call and provided as an argument on the call to `onERC20SpendableReceived` on the `ERC20SpendableReceiver`. |
+| arguments\_ | bytes   | Bytes argument to send with the call. See {mock} contracts for details on encoding and decoding arguments from bytes.                                                                |
+
+#### `IERC20SpendableReceiver`
+
+Implementation of `IERC20SpendableReceiver` interface.
+
+`ERC20Spendable` allows [ERC-20s](./eip-20.md) to operate as 'spendable' items, i.e. an [ERC-20](./eip-20.md) token that
+can trigger an action on another contract at the same time as being transfered. Similar to `ERC-677`
+and the hooks in [ERC-777](./eip-777.md), but with more of an empasis on interoperability (returned values) than
+`ERC-677` and specifically scoped interaction rather than the general hooks of [ERC-777](./eip-777.md).
+
+##### CallMustBeFromSpendableToken
+
+```solidity
+error CallMustBeFromSpendableToken()
+```
+
+Error {CallMustBeFromSpendableToken}. The call to this method can only be from a designated spendable token.
+
+##### `onERC20SpendableReceived`
+
+```solidity
+function onERC20SpendableReceived(address spender_, uint256 spent_, bytes arguments_) external returns (bytes4 retval_, bytes returnArguments_)
+```
+
+`onERC20SpendableReceived` External function called by `ERC20SpendableTokens`. This
+validates that the token is valid and then calls the internal {\_handleSpend} method.
+You must overried {\_handleSpend} in your contract to perform processing you wish to occur
+on token spend.
+
+This method will pass back the valid bytes4 selector and any bytes argument passed from
+{\_handleSpend}.
+
+###### Parameters
+
+| Name        | Type    | Description                               |
+| ----------- | ------- | ----------------------------------------- |
+| spender\_   | address | The address spending the `ERC20Spendable` |
+| spent\_     | uint256 | The amount of token spent                 |
+| arguments\_ | bytes   | Bytes sent with the call                  |
+
+## Rationale
+
+The rationale for this EIP was to remove the need for the approve -> pull interaction pattern where an EOA is performing tasks in real time. This reduces friction and the scope for exploits.
+
+## Backwards Compatibility
+
+The EIP is fully backwards compatible.
+
+## Test Cases
+
+The full SDLC for this proposal has been completed. The contract source code and unit test suite is available in `../assets/eip-7492/`, as is the source code and example (mock) implementations.
+
+## Reference Implementation
+
+Please see `../assets/eip-7492/contracts`
+
+## Security Considerations
+
+Potential negative security implications have been considered and none are envisaged. This EIP is designed to improve end-user security by the reduction in number of contract approvals required.
```

Comment:
```suggestion
Needs discussion.
```
This is a draft EIP. Unless you've got this thing audited, discussion should be solicited.